pkg/manager/internal.go

```diff
@@ -634,6 +634,7 @@ func (cm *controllerManager) startLeaderElection(ctx context.Context) (err error
 		ReleaseOnCancel: cm.leaderElectionReleaseOnCancel,
 	})
 	if err != nil {
```

Comment:
if err happend the leaderElectionStopped will not close and engageStopProcedure will hang in this code
```
     defer func() {
        // Cancel leader election only after we waited. It will os.Exit() the app for safety.
        if cm.resourceLock != nil {
	        // After asking the context to be cancelled, make sure
	        // we wait for the leader stopped channel to be closed, otherwise
	        // we might encounter race conditions between this code
	        // and the event recorder, which is used within leader election code.
	        cm.leaderElectionCancel()
	        <-cm.leaderElectionStopped
        }
  }()
```

so I think when error happened we should close this channel